src/wwwroot/app/views/home/home.controller.js

```diff
@@ -61,23 +61,17 @@ export default function HomeController( //eslint-disable-line max-statements
          vm.userNotes = notes;
          vm.notes  = cloneDeep(vm.userNotes);
       });
-      VacancyService.search(vm.vacancy).then(response => {
-         vm.totalHome = response.total;
-         vm.vacancies = response.vacancies;
-      }).catch(_onError);
       _getCurrentUser();
       _getUpcomingEvents();
       if (!vm.user) {
          $state.go('login');
       }
+      SearchService.getVacancies(vm.vacancy).then(r => set(vm, 'vacancies', r)).catch(_onError);
    }());
 
    function pageChanged(newPage) {
       vm.vacancy.current = newPage - 1;
-      VacancyService.search(vm.vacancy).then(response => {
-         vm.totalHome = response.total;
-         vm.vacancies = response.vacancies;
-      }).catch(_onError);
+      SearchService.getVacancies(vm.vacancy).then(r => set(vm, 'vacancies', r)).catch(_onError);
```

Comment:
if you name your function getVacancies it has to return promise with vacancies, but not searchResultObject. so you can name your function into searchVacancies or some other way.

as well as argument's name 'r' doesn't show the sense of this argument. 
Also your code can be next:
SearchService.getVacancies(vm.vacancy).then(partial(set(vm, 'vacancies')).catch(_onError);